### PR TITLE
Read more bytes if returned count does not match the requested length…

### PR DESCRIPTION
**Pull request by eldridgemartinez**
…. End of file also returns 0 so its non-breaking

_I certify that I own, and have sufficient rights to contribute, all source code and related material intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License._

### CODE REVIEW

**Comment by eldridgemartinez** [comment]
Please refer https://github.com/icsharpcode/SharpZipLib/issues/181 for details of this PR.

**Comment by piksel** [comment]
Don't you mean #303? #181 has nothing to do with the amount of bytes read from the input stream...

File: src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs [hunk 1]
```diff
@@ -4038,11 +4038,27 @@ public override int Read(byte[] buffer, int offset, int count)
 						baseStream_.Seek(readPos_, SeekOrigin.Begin);
 					}
 					int readCount = baseStream_.Read(buffer, offset, count);
+
+					// Azure CloudBlockBlob SDK Stream implementation is: Returns only the bytes in its buffer and does not fetch from remote until its buffer is empty.
+					// This observation causes an issue in above baseStream_.Read call because the buffer is only filled partially
+					int correctReadCount = readCount;
```
Comment (piksel) [inline src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs]: This file works according to spec. If the underlying stream returns fewer bytes, it correctly returns the same amount of bytes. The changes does not belong here (and neither does the comment about third-party specific details).

File: src/ICSharpCode.SharpZipLib/ICSharpCode.SharpZipLib.csproj [hunk 2]
```diff
@@ -10,9 +10,9 @@
   
   <!-- Nuget specific tags -->
   <PropertyGroup>
-    <AssemblyVersion>1.1.0.6</AssemblyVersion>
```
[the remark this answers is not in the document]
Comment (eldridgemartinez) [reply src/ICSharpCode.SharpZipLib/ICSharpCode.SharpZipLib.csproj]: Oh sorry. I was doing my test build and forgot to cleanup commits. I will resolve this. Thanks

File: src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs (hunk 1 above)
Comment (eldridgemartinez) [reply src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs]: I agree that the change can be applied in the caller ZipAESStream.cs file too. Thanks for looking and let me know if you applied that fix and I will help test. You can ignore this PR though.

**Comment by eldridgemartinez** [comment]
Hi, thanks for looking and pointing to where should the fix go. I agree with the fix to be applied in ZipAESStream.cs to read more and dont treat the returned lesser bytes as end of file. I will also try to see ans fix there. I badly needed this fix as we are closing to production release. Thank you

**Comment by eldridgemartinez** [comment]
hi, I close this PR and create a fresh one. I'll send in a while after I'm done testing.. I follow your suggestion to fix on ZipAESStream.cs.

**Pull request closed without merging**